src/app/Exec/lib/types/executor.js

```diff
@@ -296,14 +152,11 @@ class Executor extends Type {
         };
     }
 
-    getEnv(jobInfo) {
-        return this._convToEnvObj(jobInfo, false, 6);
-    }
-
     async allocate(job, slave) {
         this.jobId = job._id;
         this.jobName = job.name;
         this.slaveId = slave._id;
+        this.backendId = slave.backend;
```

Comment:
Shouldn't this be set at create? For example in BackendProxy#createExecutor where it is known.